DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1163,6 +1163,10 @@ public DMASTProcStatement ProcStatement()
             } else if (Check(TokenType.DM_Else)) {
                 var loc = Current().Location;
                 Whitespace();
+                if (Check(TokenType.DM_If))
+                {
+                    DMCompiler.Error(new CompilerError(loc, "Expected \"if\" or \"else\""));
```

Comment:
Use `Error()` to trigger the parser's error recovery
```suggestion
                    Error("Expected \"if\" or \"else\"");
```